astropy/coordinates/sky_coordinate.py

```diff
@@ -1702,15 +1725,38 @@ def _parse_coordinate_arg(coords, frame, units, init_kwargs):
         data = coords.data.represent_as(frame.representation)
 
         values = []  # List of values corresponding to representation attrs
+        repr_attr_name_to_drop = []
         for repr_attr_name in repr_attr_names:
             # If coords did not have an explicit distance then don't include in initializers.
             if (isinstance(coords.data, UnitSphericalRepresentation) and
                     repr_attr_name == 'distance'):
+                repr_attr_name_to_drop.append(repr_attr_name)
                 continue
 
             # Get the value from `data` in the eventual representation
             values.append(getattr(data, repr_attr_name))
 
+        # drop the ones that were skipped because they were distances
+        for nametodrop in repr_attr_name_to_drop:
+            nameidx = repr_attr_names.index(nametodrop)
+            del repr_attr_names[nameidx]
+            del units[nameidx]
+            del frame_attr_names[nameidx]
+            del repr_attr_classes[nameidx]
+
+        if coords.data.differentials:
```

Comment:
This doesn't work because it fails for things *without* differentials.  But I think ``coords.data.differentials and 's' in coord.data.differentials`` is the way to go now that you mention this.